src/main/java/com/icthh/xm/ms/entity/service/privileges/custom/FunctionCustomPrivilegesExtractor.java

```diff
@@ -40,9 +42,18 @@ public DefaultPrivilegesValue getDefaultValue() {
 
     @Override
     public List<String> toPrivilegesList(Map<String, TypeSpec> specs) {
-        return specs.values().stream().filter(it -> TRUE.equals(it.getIsApp()))
-                                      .map(TypeSpec::getKey)
-                                      .collect(toList());
+        return specs.values().stream()
+                             .map(TypeSpec::getFunctions)
+                             .filter(not(Objects::isNull))
```

Comment:
There is spec method for checking non-null values: [https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-](Objects.nonNull)